### PR TITLE
Add mssql view support

### PR DESCRIPTION
**Pull request by mortenbpost**
Hey,

PR for adding MSSQL View support - it's not polished yet but fully functional and working. I've obviously hit a few bumps in the road because it seems it wasn't originally created with the idea that it should support anything but PG?

So this PR is mainly to get an idea whether you could be interested in adding this functionality to alembic_utils and also to get some feedback if you want. 

I've also changed a few things which I believe could be bugs. The one that causes the most issues was creating a new connection in register_entities which caused a lock issue when running tests on mssql due to the fact that the migrations and this was both querying information_schema.views.

### CODE REVIEW

File: src/alembic_utils/replaceable_entity.py [hunk 1]
```diff
@@ -278,7 +283,7 @@ def register_entities(
     def compare_registered_entities(
         autogen_context, upgrade_ops, sqla_schemas: Optional[List[Optional[str]]]
     ):
-        engine = autogen_context.connection.engine
+        connection = autogen_context.connection
```
Comment (olirice) [inline src/alembic_utils/replaceable_entity.py:286]: thank you, that was a mistake

**Comment by olirice** [comment]
In addition to the alembic_utils test suite, I'm able to test postgres support on a multi-hundred entity codebase. That testing is often where subtle bugs are surfaced. Since I don't have an equivalent mysql project to test against, I couldn't be as confident in the quality of the mysql support at this point. 

That said, I don't have any immediate plans to make significant changes to the `ReplaceableEntity` interface. If you'd like to publish mysql view support as a separate package, e.g. `alembic_utils_mysql`, I'd be happy to link to it from the README and work with you to maintain compatibility. 

Longer term I'd be open to making it an optional dependency (`alembic_utils[mysql]`) or even merging back into `alembic_utils` once its been more extensively tested

**Comment by mortenbpost** [comment]
Thanks.

It makes sense. We will soon have this running on something that's closer to 50-100 entity codebase. Once I'm confident enough that it works and when I've figured out if it's something I want to maintain I'll get back to you. I will not rule out maintaining a separate package.

Thanks for creating this package :-)

**Comment by andersberliner** [comment]
@mortenbpost great work here.  I have actually forked this package @olirice to implement views in MySQL (note @mortenbpost is using MSSQL - microsoft sql server) and am using it on a database with 100s of entities.  I noticed that @olirice was making a lot of updates recently and had considered writing a PR back to this, too, to keep things in sync.

The changes required are similar to those made by @mortenbpost here - dialect specific changes to how views, etc., are created, etc..  I also noticed a slight change required in the event listener `compare_registered_entities` to support multiple databases (i.e. multiple sqlalchemy bases) from a single implementation, which is another unique characteristic of the system I am supporting.

Let me know @olirice your interest level in those kind of additions; from this conversation, a separate package similarly seems like your desire.

Let me echo my thanks for your awesome work on this package.

**Comment by olirice** [comment]
Re: mysql/mssql: whoops, thanks for the correction. Gotta work on those reading skills!

I'd be interested to hear specifics about the changes you had to make to compare_registered_entities. Ideally there would not contain anything postgres specific.

**Comment by mortenbpost** [comment]
@andersberliner I'd also be interested in the specifics related to compare registered entities. 

Just updated the branch so it's based on the most recent changes - was straight forward.

**Comment by olirice** [comment]
closing for age

**Pull request closed without merging**